### PR TITLE
Remove word restriction in corda x500 name

### PR DESCRIPTION
**Pull request by patrickkuo**
https://github.com/corda/corda/issues/2326

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt [hunk 1]
```diff
@@ -74,7 +71,6 @@ data class CordaX500Name(val commonName: String?,
     }
 
     companion object {
-        const val LENGTH_COUNTRY = 2
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: Ouch, we can't remove this as it's now public API. Just deprecate it and say it's no longer used.

Comment (patrickkuo) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: :( ok

**Comment by shamsasari** [comment]
The change log needs to be updated as well.

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -6,6 +6,8 @@ from the previous milestone release.
 
 UNRELEASED
 ----------
+* Removed blacklisted word checks in corda x500 name to allow "Server" or "Node" to be use as part of the legal name.
```
Comment (shamsasari) [inline docs/source/changelog.rst]: Corda X.500

**Pull request merged**